_health-care/_js/reducers/veteran/index.js

```diff
@@ -187,17 +187,38 @@ const blankVeteran = {
 function veteran(state = blankVeteran, action) {
   let newState = undefined;
   switch (action.type) {
-    case VETERAN_FIELD_UPDATE:
+    case VETERAN_FIELD_UPDATE: {
       newState = Object.assign({}, state);
       _.set(newState, action.propertyPath, action.value);
       return newState;
+    }
 
-    case ENSURE_FIELDS_INITIALIZED:
+    case ENSURE_FIELDS_INITIALIZED: {
       newState = Object.assign({}, state);
       // TODO(awong): HACK! Assigning to the sub object assumes pathToData() returns a reference
       // to the actual substructre such that it can be reassigned to.
       Object.assign(pathToData(newState, action.path), initializeNullValues(pathToData(newState, action.path)));
       return newState;
+    }
+
+    // Copies the veteran's address into the spouse's address fields if they have the same address.
+    // Clears the spouse's address fields if they do not.
+    case UPDATE_SPOUSE_ADDRESS: {
+      newState = Object.assign({}, state);
+      const emptyAddress = {
+        street: null,
+        city: null,
+        country: null,
+        state: null,
+        zipcode: null,
+      };
+      if (action.value) {
+        _.set(newState, action.propertyPath, state.veteranAddress.address);
+      } else {
```

Comment:
What I was thinking is if for the following situation:
1. They check they have the same address
2. Spouse address fields are filled with veteran's address
3. They decide to uncheck that they have the same address
4. The spouse address fields should be reinitialized to blank

Then, they should be able to complete the spouse address fields as normal and those would update, since this action only is called when the `sameAddress` checkbox is checked.